### DISCUSSION TITLE
Ability to purge all completed requests in a single action

### DISCUSSION THREAD

**Discussion by Weavus** [sct/overseerr#548] [Feature Requests]
I'd like to be able to delete all completed requests with a single action rather than having to do them all manually. In fact, I think having a separate 'Completed Requests' alongside the 'Recent Requests' list on the homepage would also be nice...

**Comment by sct** [comment]
May I ask why you want to purge your completed requests? Overseerr by design is meant to keep a request history, so the option to mass delete isn't available. Deletion is mainly to reset state to allow something to be requested again.

**Comment by Weavus** [comment]
I appreciate the design decision of keeping all request history so maybe what I'm after is a way to filter out completed requests or list them in a separate place as once you get a few hundred requests the single Requests lists gets a bit muddled with the mix of completed or outstanding requests in my opinion. As an admin I'm mainly concerned with what's still outstanding and I get told about completed requests by notifications anyway.

**Reply by sct** [reply]
I hear you. That’s why we have an issue for that :) https://github.com/sct/overseerr/issues/431

**Comment by lazynooblet** [comment]
I don't care for keeping request history. I have recently purged 50% of items, and wanted all the history of requests that had been fulfilled to be removed, allowing users to re-request should they please.

Running this on the SQLite database worked for me.

```SQL
DELETE FROM media_request WHERE id IN (SELECT media_request.id FROM media LEFT JOIN media_request ON media.id = media_request.mediaId WHERE media.status=5);
```

**Reply by ericvlog** [reply]
> I don't care for keeping request history. I have recently purged 50% of items, and wanted all the history of requests that had been fulfilled to be removed, allowing users to re-request should they please.
> 
> Running this on the SQLite database worked for me.
> 
> ```sql
> DELETE FROM media_request WHERE id IN (SELECT media_request.id FROM media LEFT JOIN media_request ON media.id = media_request.mediaId WHERE media.status=5);
> ```

it this working on latest overseerr ? I get in sql and execute the command seem is unable to delete now.

**Reply by obito** [reply]
if anyone is coming by again, media.status looks like it's `2` now for completed.

```
BEGIN;

DELETE FROM media_request WHERE status = 2;

DELETE FROM media
WHERE id NOT IN (SELECT mediaId FROM media_request WHERE mediaId IS NOT NULL);

COMMIT;
``` 

This will clear all completed requests as well as media linked.
